packages/ruby/lib/readme/har_request.rb

```diff
@@ -23,8 +24,28 @@ def as_json
 
     private
 
+    def request_body
+      parsed_body = JSON.parse(@request.body)
+      filter_hash_with_blacklist(parsed_body)
+    rescue
+      @request.body
+    end
+
     def to_hash_array(hash)
       hash.map { |name, value| {name: name, value: value} }
     end
+
+    def filter_hash_with_blacklist(hash)
+      hash.filter { |key, _value| !@blacklist.include?(key) }
+    end
+
+    def filter_hash_array_with_blacklist(array)
+      array.filter { |pair| !@blacklist.include? pair[:name] }
+    end
+
+    def filterd_hash_array(hash)
+      as_array = to_hash_array(hash)
+      filter_hash_array_with_blacklist(as_array)
```

Comment:
If we filter the hash before converting to an array, I think we can eliminate the need for the `filter_hash_array_with_blacklist` method?